graders/c/cgrader/cgrader.py

```diff
@@ -111,9 +111,10 @@ def run_command(self, command, input=None, sandboxed=True, timeout=None, env=Non
                 out = proc.communicate(timeout=timeout)[0]
             except subprocess.TimeoutExpired:
                 tostr = TIMEOUT_MESSAGE
-            finally:
-                out = out.decode("utf-8", "backslashreplace") if out else ""
-                return out + tostr
+
+            if out:
+                out = out.decode("utf-8", "backslashreplace")
+        return out + tostr
```

Comment:
This is a change of behavior. Is this intentional? cc @jonatanschroeder

Reply:
This is related to https://docs.astral.sh/ruff/rules/return-in-try-except-finally/. I tried to keep the behavior the same.

Reply:
I believe in general the behaviour is the same, except if an exception other than `TimeoutExpired` happens. The docs don't list any other explicit exception being raised, so this should generally be safe.

Reply:
Thanks for checking. The "what if another exception is raised" case is what I was concerned about. If that won't happen, I'm good with all this.